pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -182,6 +202,63 @@ func (c *ComponentHandler) createControllerRevision(mt metav1.Object, obj runtim
 	return true
 }
 
+func (c *ComponentHandler) cleanupControllerRevision(mt metav1.Object, curComp *v1alpha2.Component) error {
+	labels := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			ControllerRevisionComponentLabel: curComp.Name,
+		},
+	}
+	selector, err := metav1.LabelSelectorAsSelector(labels)
+	if err != nil {
+		return err
+	}
+	revisions, err := c.appsClient.ControllerRevisions(mt.GetNamespace()).List(context.Background(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Get appConfigs and workloads filter controllerRevision used
+	appConfigs := &v1alpha2.ApplicationConfigurationList{}
+	if err := c.client.List(context.Background(), appConfigs); err != nil {
```

Comment:
List all appconfig is heavy, what about mark lable on controllerrevison? If AppConfig specify using one revision, mark a lable on controllerrevison object, and this information also can find from status of AppConfig. If someday the revision is removed from AppConfig, we can compare the spec and status to remove the mark from controllerrevison object too.